types/react-onsenui/index.d.ts

```diff
@@ -3,11 +3,17 @@
 // Definitions by: Ozytis <https://ozytis.fr>,
 //                 Salim <https://github.com/salim7>,
 //                 Jemmyw <https://github.com/jemmyw>
+//                 Mikael Lirbank <https://github.com/lirbank>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
 import * as React from 'react';
-import Component = React.Component;
+
+export type Optional<T> = { [P in keyof T]?: T[P] };
+export type HTMLAttributes<K extends keyof React.HTMLAttributes<{}>> = Optional<Pick<React.HTMLAttributes<{}>, K>>;
+export type InputHTMLAttributes<K extends keyof React.InputHTMLAttributes<{}>> = Optional<Pick<React.InputHTMLAttributes<{}>, K>>;
+
+export class Component<P = {}, S = {}> extends React.Component<HTMLAttributes<'id'> & P, S> {}
```

Comment:
this looks good. I wonder if you want to add `className` as a default here also and remove it from the variously specified ones.

I'm not sure we need to export these types as they're internal use only.

Reply:
I've added `className` to the defaults list.

The exports are actually needed for the linter to pass. If you run `npm run lint react-onsenui` (or check the failed Travis run above) without exporting everything you will get:
```
ERROR: 12:6  strict-export-declare-modifiers  Prefer to explicitly write 'export' for an external module export. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 13:6  strict-export-declare-modifiers  Prefer to explicitly write 'export' for an external module export. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 14:6  strict-export-declare-modifiers  Prefer to explicitly write 'export' for an external module export. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
```

See -> https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md

> Do add the export keyword unnecessarily, because sometimes it is necessary and we want to be consistent.